tools/cdk-release/lib/lifecycles/commit.ts

```diff
@@ -0,0 +1,61 @@
+import * as path from 'path';
+import { formatCommitMessage } from '../private/format-commit-message';
+import { checkpoint } from '../private/print';
+import { runExecFile } from '../private/run-exec-file';
+import { ReleaseOptions } from '../types';
+
+export async function commit(args: ReleaseOptions, newVersion: string, bumpedFiles: string[]): Promise<void> {
+  if (args.skip?.commit) {
+    return;
+  }
+  await execCommit(args, newVersion, bumpedFiles);
+}
+
+async function execCommit (args: ReleaseOptions, newVersion: string, bumpedFiles: string[]): Promise<void> {
+  let msg = 'committing %s';
+  let paths = new Array<string>();
+  // const verify = args.verify === false || args.n ? ['--no-verify'] : [];
+  const sign = args.sign ? ['-S'] : [];
+  const toAdd = new Array<string>();
+
+  // only start with a pre-populated paths list when CHANGELOG processing is not skipped
+  if (!args.skip?.changelog) {
```

Comment:
Why isn't `bumpedFiles` just `modifiedFiles`, which already includes the changelog file if it changed?

Now the logic of what argument flags mean gets duplicated in multiple places.

Reply:
Yep, this is classic ConventionalChangelog code that I left as-is.

Simplified it according to your suggestion.